src/nnet3/nnet-discriminative-training.cc

```diff
@@ -123,16 +133,16 @@ void NnetDiscriminativeTrainer::ProcessOutputs(const NnetDiscriminativeExample &
                                kUndefined);
 
     discriminative::DiscriminativeObjectiveInfo stats(opts_.discriminative_config);
-    
+
     if (objf_info_.count(sup.name) == 0) {
       objf_info_[sup.name].stats.Configure(opts_.discriminative_config);
       objf_info_[sup.name].stats.Reset();
     }
-
+    
     ComputeDiscriminativeObjfAndDeriv(opts_.discriminative_config, 
                                       tmodel_, log_priors_,
                                       sup.supervision, nnet_output,
-                                      &(objf_info_[sup.name].stats), 
+                                      &stats, 
                                       &nnet_output_deriv,
```

Comment:
vimal, it looks to me like with this change you are just throwing away the stats.